landlab/components/soil_grading/soil_grading.py

```diff
@@ -0,0 +1,540 @@
+            The mean size of grain size in each clas.
+        alpha : float (-. 0-1), optional
+            The fraction of parent grain size that weathered to daughters.
+        A_factor : float (-. 0-1), optional
+            Factor that control the fragmentation rate.
+        soil_density : float (kg/m^3), optional
+            Density of the soil particles.
+        phi : float (-. 0-1), optional
+            Soil porosity.
+        grain_max_size : float (m), optional
+            The maximal grain size in the distribution.
+        power_of : float, optional
+            A parameter that controls the geometry relation between the grain sizes in the soil.
+        initial_median_size : float (m), optional
+            The initial median grain size in the soil.
+        initial_total_soil_weight : float (kg), optional
+            The total initial soil weight (taking into account all grain size classes).
+        std: float, optional
+            The standard deviation of grain size distribution.
+        CV: float, optional
+            Coefficient of variance of the grain size distribution.
+        is_bedrock_distribution_flag: bool, optional
+            A flag to indicate if the grain size distribution is generated for soil or
+            bedrock layer.
+        precent_of_volume_in_spread: float, optional
+            The precent of volume transferred from parent to daughter in case of 'spread'
+             grading pattern
+        """
+
+        super().__init__(grid)
+
+        self._grading_name = grading_name
+        self._meansizes = meansizes
+        self._alpha = alpha
+        self._A_factor = A_factor
+
+        if self._meansizes is not None:
+            self._n_sizes = np.size(self._meansizes)
+            self._grain_max_size = np.max(self._meansizes)
+            self._input_sizes_flag = True
+
+        else:
+            self._input_sizes_flag = False
+            self._n_sizes = n_of_grainsize_classes
+            self._grain_max_size = grain_max_size
+
+        self._N = int(self._grading_name.split("-")[0][1:])
+        self._soil_density = float(soil_density)
+        self._phi = phi
+        self._power_of = power_of
+        self._initial_median_size = initial_median_size
+        self._CV = CV
+        self._initial_total_soil_weight = initial_total_soil_weight
+        if std is None:
+            std = self._initial_median_size * self._CV
+        self._std = std
+        self._is_bedrock_distribution_flag = is_bedrock_distribution_flag
+        self._precent_of_volume_in_spread = precent_of_volume_in_spread
+
+        # Note: Landlabs' init_out_field procedure will not work
+        # for the 'grains__weight' and 'grains_classes__size' fields
+        # because the shape of these fields is: n_nodes x n_grain_sizes.
+        grid.add_field(
+            "median_size__weight",
+            np.zeros(grid.shape[0] * grid.shape[1]),
+            at="node",
+            dtype=float,
+        )
```

Comment:
This can be written more simply as,
```python
grid.add_zeros("median_size__weight", at="node")
```

and the subsequent ones as, for example,
```python
grid.at_node["grains_classes__size"] = np.ones((grid.number_of_nodes, self._n_sizes))
```

Reply:
Changed